corehq/apps/app_manager/suite_xml/post_process/remote_requests.py

```diff
@@ -254,11 +263,34 @@ def build_query_prompts(self):
 
     def build_stack(self):
         stack = Stack()
+        if module_uses_smart_links(self.module):
+            # TODO: XPath instead of string
+            frame = PushFrame(if_clause=XPath("instance('results')/results/case[@case_id=instance('commcaresession')/session/data/search_case_id]/commcare_project != instance('commcaresession')/session/user/data/commcare_project"))
+            frame.add_datum(StackJump(
+                url=Text(
+                    xpath=SuiteXPath(
```

Comment:
Good point. https://github.com/dimagi/commcare-hq/pull/30680

I renamed `xml_models.XPath` instead, to `TextXPath`, because it has way fewer occurrences, and it's a more limited class (only used inside text elements). Open to other names.